In [ ]:
import numpy as np 
import pandas as pd
import io
from matplotlib import pyplot as plt
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


In [ ]:
movies = pd.read_csv('/content/movies.csv')
movies

id    imdb_id  popularity  ...  release_year    budget_adj   revenue_adj
0     135397  tt0369610   32.985763  ...        2015.0  1.379999e+08  1.392446e+09
1      76341  tt1392190   28.419936  ...        2015.0  1.379999e+08  3.481613e+08
2     262500  tt2908446   13.112507  ...        2015.0  1.012000e+08  2.716190e+08
3     140607  tt2488496   11.173104  ...        2015.0  1.839999e+08  1.902723e+09
4     168259  tt2820852    9.335014  ...        2015.0  1.747999e+08  1.385749e+09
...      ...        ...         ...  ...           ...           ...           ...
8290   10497  tt0104779    0.529727  ...        1992.0  7.770121e+06  2.894844e+06
8291    9278  tt0104299    0.524767  ...        1992.0  0.000000e+00  0.000000e+00
8292   14002  tt0103767    0.521669  ...        1992.0  6.216097e+06  0.000000e+00
8293   10699  tt0104412    0.518807  ...        1992.0  6.526902e+07  0.000000e+00
8294   47821  tt0103976    0.499566  ...           NaN           NaN           NaN

[8295 rows x 21 columns]

In [ ]:
movies.shape

(8295, 21)

# 1) **Which are the movies with the third lowest and third highest budget?**

In [ ]:
movies.sort_values(['budget'], ascending =[False]).head()

id    imdb_id  popularity  ...  release_year    budget_adj   revenue_adj
2244  46528  tt1032751    0.250540  ...        2010.0  4.250000e+08  1.108757e+07
3375   1865  tt1298650    4.955130  ...        2011.0  3.683713e+08  9.904175e+08
7387    285  tt0449088    4.965391  ...        2007.0  3.155006e+08  1.010654e+09
14    99861  tt2395427    5.944927  ...        2015.0  2.575999e+08  1.292632e+09
6570   1452  tt0348150    1.957331  ...        2006.0  2.920507e+08  4.230205e+08

[5 rows x 21 columns]

### The Movie with third highest budget is [Pirates of the Caribbean: At World's End]




In [ ]:
df = movies.sort_values(['budget'], ascending =[True])
df

id    imdb_id  popularity  ...  release_year    budget_adj   revenue_adj
4147  21189  tt0308514    0.144924  ...        2002.0  0.000000e+00  0.000000e+00
7622   4998  tt0459102    0.575573  ...        2007.0  0.000000e+00  0.000000e+00
7623  13061  tt1134859    0.362440  ...        2007.0  0.000000e+00  0.000000e+00
7624  13470  tt0821638    0.360061  ...        2007.0  0.000000e+00  0.000000e+00
7625  14459  tt0800367    0.359607  ...        2007.0  0.000000e+00  0.000000e+00
...     ...        ...         ...  ...           ...           ...           ...
6570   1452  tt0348150    1.957331  ...        2006.0  2.920507e+08  4.230205e+08
14    99861  tt2395427    5.944927  ...        2015.0  2.575999e+08  1.292632e+09
7387    285  tt0449088    4.965391  ...        2007.0  3.155006e+08  1.010654e+09
3375   1865  tt1298650    4.955130  ...        2011.0  3.683713e+08  9.904175e+08
2244  46528  tt1032751    0.250540  ...        2010.0  4.250000e+08  1.108757e+07

[8295 rows x 21 columns]

In [ ]:
df[(df != 0).all(1)].head()

id    imdb_id  popularity  ...  release_year  budget_adj  revenue_adj
2618  39964  tt0120836    0.090186  ...        1999.0    1.309053   130.905285
3581  59296  tt1436559    0.520430  ...        2011.0    0.969398  1335.830503
2398  39356  tt1560139    0.028456  ...        2010.0    3.000000    43.000000
8235  25066  tt0114609    0.105894  ...        1995.0    8.585801    15.740636
8125  11980  tt0114194    0.705828  ...        1995.0   11.447735    22.895470

[5 rows x 21 columns]

In [ ]:
movies.budget.unique()

array([150000000, 110000000, 200000000, 190000000, 135000000, 155000000,
       108000000,  74000000, 175000000, 245000000, 176000003,  15000000,
        88000000, 280000000,  44000000,  48000000, 130000000,  95000000,
       160000000,  30000000,  40000000,  28000000,  68000000,  81000000,
        20000000,  61000000,         0,  49000000,  29000000,  58000000,
         6000000,  50000000,  11000000,  90000000,  75000000,  25000000,
        10000000,  12000000,   4000000,  11800000,  35000000,  55000000,
        60000000, 105000000,  26000000,  70000000, 120000001,   3500000,
        65000000,  50100000, 100000000,  13000000,  99000000,   2500000,
        34000000,  80000000,  17000000,  31000000,   5000000,  64000000,
        11930000,   8500000,  23000000,  53000000,   8900000,    700000,
        14800000,  19500000,   1000000,   8000000,   1800000,   2000000,
        18000000,    630019,  14000000,  37000000,   7000000,   3000000,
         9600000,   2240000,   3300000,    100000, 

In [ ]:
# Considering the budget should at least be a 4 digit number, if we think logically for a movie.

movies[movies.budget >= 5000].sort_values(['budget'], ascending =[True]).head()

id    imdb_id  popularity  ...  release_year   budget_adj    revenue_adj
1817   17632  tt1323925    0.164377  ...        2009.0  5082.002171       0.000000
2396   26214  tt1595473    0.006640  ...        2010.0  5000.000000       0.000000
7813   22887  tt0914809    0.065543  ...        2007.0  6310.011496    6310.011496
7044   14337  tt0390384    0.856329  ...        2004.0  8081.117799  490362.228024
4806  138941  tt2149360    0.159747  ...        2012.0  7597.954251       0.000000

[5 rows x 21 columns]

### Looking into the dataset yielded there are a no. of garbage value (budget = 1,2,3) which if we think logically, is impossible for a movie. Thus, taking a min. budget for every movie in the data set.
## The movie with 3rd lowest budget is [Primer] starring Shane Carruth,David Sullivan, Casey Gooden.


# **2) Which are the movies with most and least earned revenue?**

In [ ]:
tx = movies.sort_values(['revenue'], ascending =[False]).head()
tx

id    imdb_id  popularity  ...  release_year    budget_adj   revenue_adj
1386   19995  tt0499549    9.432768  ...        2009.0  2.408869e+08  2.827124e+09
3     140607  tt2488496   11.173104  ...        2015.0  1.839999e+08  1.902723e+09
5231     597  tt0120338    4.355219  ...        1997.0  2.716921e+08  2.506406e+09
4361   24428  tt0848228    7.637767  ...        2012.0  2.089437e+08  1.443191e+09
0     135397  tt0369610   32.985763  ...        2015.0  1.379999e+08  1.392446e+09

[5 rows x 21 columns]

### The movie with highest earned revenue is [Avatar] with 2781505847 dollars grossed worldwide.

In [ ]:
tx = movies.sort_values(['revenue'], ascending =[True])
tx

id    imdb_id  popularity  ...  release_year    budget_adj   revenue_adj
4147   21189  tt0308514    0.144924  ...        2002.0  0.000000e+00  0.000000e+00
3702   66125  tt0803061    0.302890  ...        2011.0  0.000000e+00  0.000000e+00
3701   66945  tt1519663    0.303053  ...        2011.0  0.000000e+00  0.000000e+00
3700   77887  tt1850374    0.303240  ...        2011.0  0.000000e+00  0.000000e+00
3699   58625  tt1707687    0.306280  ...        2011.0  0.000000e+00  0.000000e+00
...      ...        ...         ...  ...           ...           ...           ...
0     135397  tt0369610   32.985763  ...        2015.0  1.379999e+08  1.392446e+09
4361   24428  tt0848228    7.637767  ...        2012.0  2.089437e+08  1.443191e+09
5231     597  tt0120338    4.355219  ...        1997.0  2.716921e+08  2.506406e+09
3     140607  tt2488496   11.173104  ...        2015.0  1.839999e+08  1.902723e+09
1386   19995  tt0499549    9.432768  ...        2009.0  2.408869e+08  2.827124e+09

[8295 rows x 21 columns]

In [ ]:
movies.revenue.unique()

array([1513528810,  378436354,  295238201, ...,   88036683,    1862805,
         14683921])

In [ ]:
tx[(tx != 0).all(1)]

id    imdb_id  popularity  ...  release_year    budget_adj   revenue_adj
5067   13537  tt0323944    0.462609  ...        2003.0  7.112116e+06  2.370705e+00
8142    2293  tt0113749    0.552091  ...        1995.0  8.585801e+06  2.861934e+00
3239   14301  tt1227926    0.352054  ...        2008.0  2.025573e+05  3.038360e+00
5162   24914  tt0368574    0.208637  ...        2003.0  1.185353e+01  5.926763e+00
8226   48781  tt0113965    0.183034  ...        1995.0  9.158188e+06  8.585801e+00
...      ...        ...         ...  ...           ...           ...           ...
0     135397  tt0369610   32.985763  ...        2015.0  1.379999e+08  1.392446e+09
4361   24428  tt0848228    7.637767  ...        2012.0  2.089437e+08  1.443191e+09
5231     597  tt0120338    4.355219  ...        1997.0  2.716921e+08  2.506406e+09
3     140607  tt2488496   11.173104  ...        2015.0  1.839999e+08  1.902723e+09
1386   19995  tt0499549    9.432768  ...        2009.0  2.408869e+08  2.827124e+09

[2958 rows x 21 columns]

### The movie with least earned revenue is [Manos: The Hands of Fate] with 0 revenue.

### If we ignore the 0 values, then the movie with least earned revenue is [Mallrats] with 2



# 3) **What is the most common Genre for Vin Diesel & Emma Watson movies?**

In [ ]:
movies[(movies.cast.str.contains('Vin Diesel') == True) & (movies.cast.str.contains('Emma Watson') == True)].genres

Series([], Name: genres, dtype: object)

In [ ]:
a = movies[movies.cast.str.contains('Emma Watson') == True].genres.values
a

array(['Horror|Mystery|Thriller', 'Drama|Adventure',
       'Adventure|Fantasy|Family', 'Adventure|Fantasy|Family',
       'Adventure|Fantasy|Family', 'Animation|Comedy|Family',
       'Adventure|Family|Fantasy', 'Adventure|Fantasy|Family',
       'Drama|Romance', 'Drama|Crime', 'Adventure|Fantasy|Family',
       'Adventure|Fantasy|Family', 'Adventure|Fantasy|Family|Mystery',
       'Drama|Family'], dtype=object)

In [ ]:
b = movies[movies.cast.str.contains('Vin Diesel') == True].genres.values
b

array(['Action|Crime|Thriller', 'Fantasy|Action|Adventure',
       'Action|Science Fiction|Adventure', 'Action|Crime|Drama|Thriller',
       'Action|Drama',
       'Adventure|Animation|Family|Fantasy|Science Fiction',
       'Action|Crime|Thriller', 'Crime|Thriller',
       'Action|Adventure|Science Fiction|Thriller',
       'Action|Thriller|Crime', 'Action', 'Action|Drama', 'Action',
       'Science Fiction|Action|Thriller', 'Action|Thriller|Crime',
       'Action|Animation|Science Fiction',
       'Action|Comedy|Drama|Family|Thriller', 'Drama|Action|Comedy|Crime',
       'Adventure|Action|Thriller|Science Fiction',
       'Action|Animation|Science Fiction|Thriller'], dtype=object)

In [ ]:
Emma = ['Horror','Mystery','Thriller', 'Drama','Adventure',
       'Adventure','Fantasy','Family', 'Adventure','Fantasy','Family',
       'Adventure','Fantasy','Family', 'Animation','Comedy','Family',
       'Adventure','Family','Fantasy', 'Adventure','Fantasy','Family',
       'Drama','Romance', 'Drama','Crime', 'Adventure','Fantasy','Family',
       'Adventure','Fantasy','Family', 'Adventure','Fantasy','Family','Mystery',
       'Drama','Family']

Vin = ['Action','Crime','Thriller', 'Fantasy','Action','Adventure',
       'Action','Science Fiction','Adventure', 'Action','Crime','Drama','Thriller',
       'Action','Drama',
       'Adventure','Animation','Family','Fantasy','Science Fiction',
       'Action','Crime','Thriller', 'Crime','Thriller',
       'Action','Adventure','Science Fiction','Thriller',
       'Action','Thriller','Crime', 'Action', 'Action','Drama', 'Action',
       'Science Fiction','Action','Thriller', 'Action','Thriller','Crime',
       'Action','Animation','Science Fiction',
       'Action','Comedy','Drama','Family','Thriller','Drama','Action','Comedy','Crime',
       'Adventure','Action','Thriller','Science Fiction',
       'Action','Animation','Science Fiction','Thriller']

In [ ]:
list(set(Emma).intersection(set(Vin)))

['Family',
 'Thriller',
 'Comedy',
 'Drama',
 'Animation',
 'Adventure',
 'Fantasy',
 'Crime']

In [ ]:
from collections import Counter

c = list((Counter(Emma) & Counter(Vin)).elements())
c

['Thriller',
 'Drama',
 'Drama',
 'Drama',
 'Drama',
 'Adventure',
 'Adventure',
 'Adventure',
 'Adventure',
 'Adventure',
 'Fantasy',
 'Fantasy',
 'Family',
 'Family',
 'Animation',
 'Comedy',
 'Crime']

### The most common Genre for Vin Diesel & Emma Watson movies is Adventure

# 4) **Name any 3 production companies which have invested money in worse revenue movies**

In [ ]:
tx = movies.sort_values(['revenue'], ascending =[True])
tx

id    imdb_id  popularity  ...  release_year    budget_adj   revenue_adj
4147   21189  tt0308514    0.144924  ...        2002.0  0.000000e+00  0.000000e+00
3702   66125  tt0803061    0.302890  ...        2011.0  0.000000e+00  0.000000e+00
3701   66945  tt1519663    0.303053  ...        2011.0  0.000000e+00  0.000000e+00
3700   77887  tt1850374    0.303240  ...        2011.0  0.000000e+00  0.000000e+00
3699   58625  tt1707687    0.306280  ...        2011.0  0.000000e+00  0.000000e+00
...      ...        ...         ...  ...           ...           ...           ...
0     135397  tt0369610   32.985763  ...        2015.0  1.379999e+08  1.392446e+09
4361   24428  tt0848228    7.637767  ...        2012.0  2.089437e+08  1.443191e+09
5231     597  tt0120338    4.355219  ...        1997.0  2.716921e+08  2.506406e+09
3     140607  tt2488496   11.173104  ...        2015.0  1.839999e+08  1.902723e+09
1386   19995  tt0499549    9.432768  ...        2009.0  2.408869e+08  2.827124e+09

[8295 rows x 21 columns]

### Quixote Films, Low Key Productions, Screen Australia,Endymion Films, Lions Gate Films are some of the production companies which have invested money in worse revenue movies



#**5) What is the average runtime of movies in the year 2006?**

In [ ]:
movies[movies.release_year.isnull()]

id    imdb_id  popularity  ...  release_year  budget_adj revenue_adj
8294  47821  tt0103976    0.499566  ...           NaN         NaN         NaN

[1 rows x 21 columns]

In [ ]:
movies[movies.release_year == 2006].runtime.values

array([106., 151., 117., 144., 149., 104.,  91., 126., 109., 151., 136.,
       130., 108., 143., 109.,  97., 154.,  82., 107.,  75., 106., 129.,
        83., 147., 143., 122.,  88., 110., 120.,  93., 125., 105.,  96.,
       108., 102.,  97.,  85.,  94., 126., 104.,  99.,  83.,  91.,  98.,
        97., 139.,  88., 107.,  78., 103., 167.,  88., 103., 134.,  87.,
       125., 132., 101., 112.,  86., 102., 109., 117.,  99., 118., 124.,
       110.,  93., 110.,  96., 125.,  95.,  99., 114.,  97., 116.,  93.,
       117., 123., 113.,  94.,  93., 117., 121., 108.,  84.,  92., 121.,
        80.,  99., 107.,  95.,  73., 122., 108., 102.,  98.,  69.,  73.,
       101., 103., 110., 141.,  77., 100.,  90., 113., 128.,  98., 118.,
        94., 125., 112.,  95.,  83.,  93.,  94.,  83., 126.,  90.,  98.,
       120., 134., 105., 116.,  88.,  95., 139.,  90., 144., 104., 105.,
        92.,  90.,  72., 116., 114.,  41., 105.,  91., 116.,  92.,  91.,
        83., 116.,  98.,  93., 110., 106., 110., 13

In [ ]:
a = np.array([106., 151., 117., 144., 149., 104.,  91., 126., 109., 151., 136.,
       130., 108., 143., 109.,  97., 154.,  82., 107.,  75., 106., 129.,
        83., 147., 143., 122.,  88., 110., 120.,  93., 125., 105.,  96.,
       108., 102.,  97.,  85.,  94., 126., 104.,  99.,  83.,  91.,  98.,
        97., 139.,  88., 107.,  78., 103., 167.,  88., 103., 134.,  87.,
       125., 132., 101., 112.,  86., 102., 109., 117.,  99., 118., 124.,
       110.,  93., 110.,  96., 125.,  95.,  99., 114.,  97., 116.,  93.,
       117., 123., 113.,  94.,  93., 117., 121., 108.,  84.,  92., 121.,
        80.,  99., 107.,  95.,  73., 122., 108., 102.,  98.,  69.,  73.,
       101., 103., 110., 141.,  77., 100.,  90., 113., 128.,  98., 118.,
        94., 125., 112.,  95.,  83.,  93.,  94.,  83., 126.,  90.,  98.,
       120., 134., 105., 116.,  88.,  95., 139.,  90., 144., 104., 105.,
        92.,  90.,  72., 116., 114.,  41., 105.,  91., 116.,  92.,  91.,
        83., 116.,  98.,  93., 110., 106., 110., 136.,  80., 101.,  97.,
       108.,  83.,  95.,  75.,   7.,  92., 104.,  92.,  91., 126., 124.,
        86., 105.,  96., 101., 112., 115.,  92., 111.,  89.,  87.,  99.,
       103.,  98., 108.,  96., 108., 140.,   5.,  95., 122.,   7., 129.,
        90., 108., 106., 115., 111., 102.,  91., 110.,  88.,  96.,  91.,
       102., 107., 104., 105., 115.,  81., 114.,  79.,   7.,  88.,  96.,
       145.,  93., 126.,  72.,  91., 115., 103.,  97.,  73.,  96.,  76.,
       125.,  80., 110.,  85.,  75.,  88.,  11.,  87., 110.,  92., 110.,
        91., 147.,  84., 104., 118., 116.,  86.,  90.,  91.,  99., 105.,
       107., 102., 101., 110.,  90., 120.,  92.,  76.,  91.,  97.,  85.,
       125., 102.,  87.,  88.,  91.,  88., 103.,  98.,  92.,  88., 117.,
       154., 103.,  93., 111.,  85.,  90.,  90., 103.,  84., 185.,  71.,
       270., 110., 105., 109., 107.,  92.,  90., 122.,  77., 100.,  84.,
        89., 172.,  94.,  95.,  96.,  84., 103., 118.,  88.,  83.,  95.,
       101.,  87.,  90.,  98., 102.,  86., 100., 104., 108.,  94., 128.,
        99.,  74.,  89.,  78., 101.,  91.,  95.,  87.,  32., 100., 125.,
        85.,  94.,  90., 105.,  90.,  93.,  94.,  88.,  95.,  90., 104.,
       110.,  90.,  85.,  96.,  96., 107.,  92.,  93., 120.,  23., 550.,
        93., 112.,  93.,  84., 102.,  90.,  93.,  90.,  90.,  96.,  75.,
        91.,  75.,  90., 101.,  98., 145., 120.,  97.,  95.,  91., 169.,
        87.,  91.,  96.,  87.,  73., 101.,  91.,  93.,  86.,  85.,  86.,
       110.,  95.,   3.,  98., 101.,  93.,   0.,  63., 132.,  85.,  90.,
        87.,  93., 202., 144.,  99.,  97.,  85.,  88.,  94.,  81.,  92.,
        99., 121.,  85.,  14., 112., 131., 158., 107., 100.,  98.,  92.,
       135.])

In [ ]:
print(a.mean())

101.68382352941177


### The average runtime of movies in the year 2006 is 101.6838 ie., around a 100 minutes

# **6) What is the average number of words in movie titles between the year 2000-2005?**

In [ ]:
movies.columns

Index(['id', 'imdb_id', 'popularity', 'budget', 'revenue', 'original_title',
       'cast', 'homepage', 'director', 'tagline', 'keywords', 'overview',
       'runtime', 'genres', 'production_companies', 'release_date',
       'vote_count', 'vote_average', 'release_year', 'budget_adj',
       'revenue_adj'],
      dtype='object')

In [ ]:
movies[(movies.release_year >= 2000) & (movies.release_year <= 2005)].loc[:,['original_title','release_year']].original_title

2633    The Lord of the Rings: The Fellowship of the Ring
2634             Harry Potter and the Philosopher's Stone
2635                                     Mulholland Drive
2636                                         Donnie Darko
2637                                       Monsters, Inc.
                              ...                        
7264                                                 DiG!
7265                          Cazuza - O tempo não pára
7266                             In questo mondo di ladri
7267                                       Long Way Round
7268                                   Born into Brothels
Name: original_title, Length: 1460, dtype: object

In [ ]:
temp = movies[(movies.release_year >= 2000) & (movies.release_year <= 2005)].loc[:,['original_title','release_year']].original_title.values
temp

array(['The Lord of the Rings: The Fellowship of the Ring',
       "Harry Potter and the Philosopher's Stone", 'Mulholland Drive',
       ..., 'In questo mondo di ladri', 'Long Way Round',
       'Born into Brothels'], dtype=object)

In [141]:
for i in temp:
  print(len(i))

49
40
16
12
14
28
5
21
14
12
17
25
23
16
14
20
13
18
20
4
18
15
14
15
9
22
13
11
12
10
25
30
11
11
11
14
18
8
9
11
24
40
9
33
11
12
14
18
16
14
21
9
15
19
14
19
8
12
7
11
8
8
8
9
29
21
9
26
5
19
8
16
15
31
7
59
10
15
17
18
11
15
10
1
8
9
11
13
3
16
9
9
14
8
41
11
20
14
8
13
7
6
12
23
19
4
38
35
14
14
16
30
24
7
13
24
7
16
15
6
24
12
19
26
5
12
14
8
10
25
12
7
10
16
20
13
15
13
17
6
8
8
35
19
16
8
20
10
13
20
14
8
18
20
7
22
13
15
14
15
12
23
9
6
5
44
38
12
21
10
12
12
16
3
6
11
16
12
12
15
10
4
8
12
20
3
30
17
13
23
26
8
9
11
10
19
5
38
17
21
35
11
28
13
14
12
18
29
15
10
15
10
9
13
14
58
5
18
2
21
9
8
4
4
14
5
30
35
32
18
34
18
8
14
5
25
21
6
11
16
14
8
37
39
7
19
13
10
44
19
15
11
15
13
15
3
15
15
16
17
8
17
10
15
31
6
9
20
31
16
5
8
12
18
24
11
30
17
11
18
11
13
9
17
27
18
8
10
10
10
32
5
9
17
5
21
9
8
13
20
14
16
18
13
5
17
11
18
10
9
16
10
18
8
11
6
39
21
18
10
14
16
20
10
19
17
19
33
49
6
17
20
17
25
8
13
13
7
14
12
18
15
15
8
13
8
6
11
14
10
13
10
10
7
18
17
18
20
21
11
22
9
9
1

In [142]:
sum = 0
for i in temp:
  sum = sum + len(i)
sum = sum/len(temp)
sum
  

16.21232876712329

### The average no. of words in movie titles from 2000 - 2005 is 16.2123 ie., around 16 words